kv/kv.go

```diff
@@ -339,59 +340,13 @@ type Driver interface {
 	Open(path string) (Storage, error)
 }
 
-// TransactionOption indicates the option when beginning a transaction
-// `TxnScope` must be set for each object
-// Every other fields are optional, but currently at most one of them can be set
-type TransactionOption struct {
-	TxnScope   string
-	StartTS    *uint64
-	PrevSec    *uint64
-	MinStartTS *uint64
-	MaxPrevSec *uint64
-}
-
-// DefaultTransactionOption creates a default TransactionOption, ie. Work in GlobalTxnScope and get start ts when got used
-func DefaultTransactionOption() TransactionOption {
-	return TransactionOption{TxnScope: oracle.GlobalTxnScope}
-}
-
-// SetMaxPrevSec set maxPrevSec
-func (to TransactionOption) SetMaxPrevSec(maxPrevSec uint64) TransactionOption {
-	to.MaxPrevSec = &maxPrevSec
-	return to
-}
-
-// SetMinStartTS set minStartTS
-func (to TransactionOption) SetMinStartTS(minStartTS uint64) TransactionOption {
-	to.MinStartTS = &minStartTS
-	return to
-}
-
-// SetStartTs set startTS
-func (to TransactionOption) SetStartTs(startTS uint64) TransactionOption {
-	to.StartTS = &startTS
-	return to
-}
-
-// SetPrevSec set prevSec
-func (to TransactionOption) SetPrevSec(prevSec uint64) TransactionOption {
-	to.PrevSec = &prevSec
-	return to
-}
-
-// SetTxnScope set txnScope
-func (to TransactionOption) SetTxnScope(txnScope string) TransactionOption {
-	to.TxnScope = txnScope
-	return to
-}
-
 // Storage defines the interface for storage.
 // Isolation should be at least SI(SNAPSHOT ISOLATION)
 type Storage interface {
 	// Begin a global transaction
 	Begin() (Transaction, error)
 	// Begin a transaction with given option
-	BeginWithOption(option TransactionOption) (Transaction, error)
+	BeginWithOption(option tikv.StartTSOption) (Transaction, error)
```

Comment:
I think it is not a good idea to import `tikv` package in the interface Storage, Maybe define the `option` as an `interface{}` will be better since the `Storage` maybe not `tikv`, and we can convert it to `tikv.StartTSOption` in `tikv` driver.